src/de.rs

```diff
@@ -579,9 +579,9 @@ impl<S: Read> de::Decoder for Decoder<S> {
             self.decode_map(visitor).await
         } else if self.numeric.contains(&self.buffer[0]) {
             self.decode_number(visitor).await
-        } else if self.buffer.len() >= FALSE.len() && self.buffer.starts_with(FALSE) {
-            self.decode_bool(visitor).await
-        } else if self.buffer.len() >= TRUE.len() && self.buffer.starts_with(TRUE) {
+        } else if (self.buffer.len() >= FALSE.len() && self.buffer.starts_with(FALSE))
+            || (self.buffer.len() >= TRUE.len() && self.buffer.starts_with(TRUE))
+        {
             self.decode_bool(visitor).await
```

Comment:
should maybe make sure a test covers this, but clippy complained that these two condition blocks were the same, so i combined the conditions